typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -185,40 +201,102 @@ export class EvmERC20WarpRouteReader extends HyperlaneReader {
     };
   }
 
+  async fetchXERC20Config(
+    type: TokenType.XERC20 | TokenType.XERC20Lockbox,
+    xERC20Address: Address,
+    warpRouteAddress: Address,
+  ): Promise<XERC20TokenMetadata | {}> {
+    // fetch the limits if possible
+    const rateLimitsABI = [
+      'function rateLimitPerSecond(address) external view returns (uint128)',
+      'function bufferCap(address) external view returns (uint112)',
+    ];
+    const xERC20 = new Contract(xERC20Address, rateLimitsABI, this.provider);
+    try {
+      return {
+        xERC20: {
+          limits: {
+            rateLimitPerSecond: (
+              await xERC20.rateLimitPerSecond(warpRouteAddress)
+            ).toString(),
+            bufferCap: (await xERC20.bufferCap(warpRouteAddress)).toString(),
+          },
+        },
+      };
+    } catch (_error) {
+      return {};
+    }
+  }
+
   /**
    * Fetches the metadata for a token address.
    *
-   * @param tokenAddress - The address of the token.
+   * @param warpRouteAddress - The address of the token.
    * @returns A partial ERC20 metadata object containing the token name, symbol, total supply, and decimals.
    * Throws if unsupported token type
    */
   async fetchTokenConfig(
     type: TokenType,
-    tokenAddress: Address,
+    warpRouteAddress: Address,
   ): Promise<HypTokenConfig> {
     if (
       type === TokenType.collateral ||
       type === TokenType.collateralVault ||
-      type === TokenType.collateralVaultRebase
+      type === TokenType.collateralVaultRebase ||
+      type === TokenType.XERC20 ||
+      type === TokenType.XERC20Lockbox
     ) {
-      const erc20 = HypERC20Collateral__factory.connect(
-        tokenAddress,
-        this.provider,
-      );
-      const token = await erc20.wrappedToken();
+      let xerc20Token: Address | undefined;
+      let lockbox: Address | undefined;
+      let token: Address;
+      let xERC20Metadata: XERC20TokenMetadata | {} = {};
+
+      if (type === TokenType.XERC20Lockbox) {
+        // XERC20Lockbox is a special case of collateral, we will fetch it from the xerc20 contract
+        const hypXERC20Lockbox = HypXERC20Lockbox__factory.connect(
+          warpRouteAddress,
+          this.provider,
+        );
+        xerc20Token = await hypXERC20Lockbox.xERC20();
+        token = xerc20Token;
+        lockbox = await hypXERC20Lockbox.lockbox();
+      } else {
+        const erc20 = HypERC20Collateral__factory.connect(
+          warpRouteAddress,
+          this.provider,
+        );
+        token = await erc20.wrappedToken();
+      }
```

Comment:
just noting this is duped with the checker's `getCollateralToken`

Reply:
I think it is very similar, but slightly different, no? Here the input is the `warpRouteAddress`, while in the checker the input is the `token` address, right?

Reply:
I guess, though you can get the token from `.wrappedToken`

Reply:
Well in this case, we actaully want the xerc20 token when the type is XERC20Lockbox, not the "inner" token